main.c

```diff
@@ -466,13 +477,36 @@ static void setup_property_changed_listener(void) {
 }
 #endif
 
+static void seat_handle_capabilities(void *data, struct wl_seat *seat,
+		uint32_t capabilities
+) {
+	struct seat *seat_tmp = data;
+	seat_tmp->capabilities = capabilities;
+}
+
+static void seat_handle_name(void *data, struct wl_seat *seat,
+		const char *name
+) {
+	struct seat *seat_tmp = data;
+	seat_tmp->name = strdup(name);
+}
+
+static const struct wl_seat_listener wl_seat_listener = {
+	.name = seat_handle_name,
+	.capabilities = seat_handle_capabilities
```

Comment:
Style nit: add a final comma to all lines (even the last one) in a multi-line initializer.